modules/top-level/output.nix

```diff
@@ -273,25 +273,35 @@ in
         }
       );
     in
-    {
-      finalPackage = wrappedNeovim;
-      initPath = "${init}";
-
-      printInitPackage = pkgs.writeShellApplication {
-        name = "nixvim-print-init";
-        runtimeInputs = [ pkgs.bat ];
-        text = ''
-          bat --language=lua "${textInit}"
+    lib.mkMerge [
+      {
+        finalPackage = wrappedNeovim;
+        initPath = "${init}";
+
+        printInitPackage = pkgs.writeShellApplication {
+          name = "nixvim-print-init";
+          runtimeInputs = [ pkgs.bat ];
+          text = ''
+            bat --language=lua "${textInit}"
+          '';
+        };
+
+        # Add a global table at start of init
+        extraConfigLuaPre = lib.mkOrder 1 ''
+          local M = {}
         '';
-      };
 
-      extraConfigLuaPre = lib.mkIf config.wrapRc ''
-        -- Ignore the user lua configuration
-        vim.opt.runtimepath:remove(vim.fn.stdpath('config'))              -- ~/.config/nvim
-        vim.opt.runtimepath:remove(vim.fn.stdpath('config') .. "/after")  -- ~/.config/nvim/after
-        vim.opt.runtimepath:remove(vim.fn.stdpath('data') .. "/site")     -- ~/.local/share/nvim/site
-      '';
+        extraPlugins = lib.mkIf config.wrapRc [ config.filesPlugin ];
+      }
+      (lib.mkIf config.wrapRc {
```

Comment:
does this need to be elevated to top, as well?

Reply:
![image](https://github.com/user-attachments/assets/d64f9158-6b02-4912-8c61-9cfc3b2c16c9)